### PR TITLE
Remove euclid dependency from dc_bundle

### PR DESCRIPTION
**Pull request by yiqunw700**
Fixes: #1642

Migrate to proto defined LayoutTransform and AffineTransform.
Copied a few transform functions from euclid for LayoutTransform and AffineTransform.

### CODE REVIEW

File: designcompose/src/main/java/com/android/designcompose/Utils.kt [hunk 1]
```diff
@@ -1276,41 +1278,53 @@ internal constructor(
...
-            } else {
-                val skMatrix = Matrix()
-                skMatrix.setValues(
-                    floatArrayOf(it[0], it[1], it[4], it[2], it[3], it[5], 0.0f, 0.0f, 1.0f)
-                )
-                skMatrix
-            }
+            val skMatrix = Matrix()
+            skMatrix.setValues(
+                floatArrayOf(it.m11, it.m12, it.m31, it.m21, it.m22, it.m32, 0.0f, 0.0f, 1.0f)
```
Comment (rylin8) [inline designcompose/src/main/java/com/android/designcompose/Utils.kt:1325]: How did you do this conversion? I would have thought it woudl be `m11, m12, m21, m13, m14, m22, 0.0f, 0.0f, 1.0f`

Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/Utils.kt:1325]: AffineTransform has 
m11, m12,
m21, m22,
m31, m32
and it used to be it[0], it[1], it[4], it[2], it[3], it[5]

I don't know what it really does though...

**Review by rylin8 — APPROVED** [review]

File: crates/dc_bundle/src/definition/modifier.rs [hunk 2]
```diff
@@ -15,3 +15,380 @@
  */
 
 include!(concat!(env!("OUT_DIR"), "/designcompose.definition.modifier.rs"));
+
+/// Implementations are copied from euclid Transform3D.
```
Comment (iamralpht) [inline crates/dc_bundle/src/definition/modifier.rs:19]: Please put this in its own file, and include the license header from euclid.

Comment (yiqunw700) [reply crates/dc_bundle/src/definition/modifier.rs]: Done.

**Review by iamralpht — APPROVED** [review]
LGTM, thanks for making the licensing clear.

**Pull request merged**